agent/lib_aws_sdk_php.h

```diff
@@ -20,4 +33,73 @@ extern void nr_lib_aws_sdk_php_handle_version();
 extern void nr_lib_aws_sdk_php_add_supportability_service_metric(
     const char* service_name);
 
+#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO /* PHP8.1+ */
+/* Aside from service class and version detection, instrumentation is only
+ * supported with PHP 8.1+ */
+
+/*
+ * Purpose : Parses the QueueUrl to extract cloud_region, cloud_account_id, and
+ * destination_name.  The extraction sets all or none since the values are from
+ * the same string and if it is malformed, it cannot be used.
+ *
+ * Params  : 1. The QueueUrl
+ *           2. message_params to set message_params.destination_name
+ *           3. cloud_attrs to set message_params.cloud_region,
+ * message_params.cloud_account_id
+ *
+ * Returns :
+ *
+ * Note: caller is responsible for
+ * freeing cloud_attrs.cloud_region, cloud_attrs.cloud_account_id, and
+ * message_params.destination_name
+ */
+extern void nr_lib_aws_sdk_php_sqs_parse_queueurl(
+    const char* sqs_queueurl,
+    nr_segment_message_params_t* message_params,
+    nr_segment_cloud_attrs_t* cloud_attrs);
+
+/*
+ * Purpose : Handle when an SqsClient initiates a command
+ *
+ * Params  : 1. segment : if we instrument the commandName, we'll need to end
+ * the segment as a message segment
+ *           2. command_name_string : the string of the command being called
+ *           3. NR_EXECUTE_ORIG_ARGS (execute_data, func_return_value)
+ * Returns :
+ *
+ */
+extern void nr_lib_aws_sdk_php_sqs_handle(nr_segment_t* segment,
+                                          char* command_name_string,
+                                          NR_EXECUTE_PROTO);
+
+/*
+ * Purpose : The second argument to the Aws/AwsClient::__call function should be
+ * an array containing an array of argument name:value pairs. Given an argument
+ * name, this will return the value of the argument.
+ *
+ * Params  : 1. arg_name: name of argument to extract from command arg array
+ *           2. NR_EXECUTE_PROTO (execute_data, func_return_value)
+ *
+ * Returns : the value of the arg_name; NULL if does not exist
+ *
+ * Note: The caller is responsible for freeing the returned string value
+ *
+ */
+extern char* nr_lib_aws_sdk_php_get_command_arg_value(char* command_arg_name,
+                                                      NR_EXECUTE_PROTO);
+
+/*
+ * Purpose : Return the first argument to the Aws/AwsClient::__call function
+ * which should be the name of the command.
+ *
+ * Params  : 1. NR_EXECUTE_PROTO (execute_data, func_return_value)
+ *
+ * Returns : the value of the command_name; NULL if does not exist
+ *
+ * Note: The caller is responsible for freeing the returned string value
+ *
+ */
+extern char* nr_lib_aws_sdk_php_get_command_name(NR_EXECUTE_PROTO);
```

Comment:
Could this return `zend_string *` instead of raw `char *`? See [this comment](https://github.com/newrelic/newrelic-php-agent/pull/1003/files#r1909385990).

Reply:
See comment: https://github.com/newrelic/newrelic-php-agent/pull/1003#discussion_r1909652756